### PR TITLE
TEAM TEACHING - rendering avatar models as payload parts

### PR DESCRIPTION
**Pull request by ZappoMan**
This implements the adding and removing of model submesh parts as individual payload items for the skeleton and head/face models. It mostly works. But we're still seeing rendering bugs where some meshes just don't seem to render.

Other known issues:
1) changing the avatar and/or some avatar seem to be scaled wrong. (Jim? others)
2) not all avatar parts correctly render sometimes (Judd's face parts randomly don't draw.) This seems like the same issue we're seeing with entity model parts

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9e7f60e6a36ac12ed27425c369574d5a81a248d8 available at: https://gist.github.com/e26a51210bde24736ce4<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5030.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5030.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2a2be789e4ad72401969c6680d638e90c2304ad8 available at: https://gist.github.com/e26a51210bde24736ce4<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5030.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5030.dmg

**Comment by ZappoMan** [comment]
Note to self: is there a connection between the super small avatars and the invisible parts? Maybe they aren't invisible, just really small?!

File: libraries/entities-renderer/src/EntityTreeRenderer.cpp [hunk 1]
```diff
@@ -501,7 +501,7 @@ void EntityTreeRenderer::render(RenderArgs* renderArgs) {
             _viewState->getShadowViewFrustum() : _viewState->getCurrentViewFrustum();
         
         // Setup batch transform matrices
-        gpu::Batch batch;
+        gpu::Batch batch; // FIX ME - this is very suspicious!
```
Comment (samcake) [inline libraries/entities-renderer/src/EntityTreeRenderer.cpp]: This is OK, we are not rendering anything anymore (except for web entities and polymesh) though this code path.
Their rendering will migrate to the payload technique and everything will be green
That code here will get remove
All good

File: libraries/entities-renderer/src/EntityTreeRenderer.cpp [hunk 2]
```diff
@@ -525,6 +525,7 @@ void EntityTreeRenderer::render(RenderArgs* renderArgs) {
         Model::endScene(renderArgs);
         _tree->unlock();
         
+        // FIX ME - this is very suspicious!
```
Comment (samcake) [inline libraries/entities-renderer/src/EntityTreeRenderer.cpp]: Actually yes, this was me commenting out this batch execution to make sure i would avoid any side effects
As stated above, all green, this code will be useless as we move everything to the engine / payload approach

**Pull request merged**